### PR TITLE
[Fixed issue#1218 V2] Show the best metric curve during search progress in webui

### PR DESCRIPTION
**Pull request by lvybriage**
#1218 
1. when user set `optimize_mode: maximize`, best metric graph is: 
image

 and default metric graph is:
image

2. when user set `optimize_mode: minimize`, best metric graph is: 
image

3. the filed `optimize_mode: minimize |  maximize` is from tuner.classArgs.optimize_mode

### CODE REVIEW

**Comment by lvybriage** [comment]
@yangou1988 Could you help us choose the color for best curve? Now it's both blue.
image

File: src/webui/src/components/TrialsDetail.tsx [hunk 1]
```diff
@@ -336,12 +337,24 @@ class TrialsDetail extends React.Component<TrialsDetailProps, TrialDetailState>
                     let expLogCollection: boolean = false;
                     const isMultiy: boolean = res.data.params.multiPhase !== undefined
                         ? res.data.params.multiPhase : false;
+                    const tuner = res.data.params.tuner;
+                    // I'll set optimize is maximize if user not set optimize
+                    let optimize: string = 'maximize';
+                    if (tuner !== undefined) {
+                        if (tuner.classArgs !== undefined) {
+                            if (tuner.classArgs.optimize_mode !== undefined) {
+                                if (tuner.classArgs.optimize_mode !== 'maximize') {
```
Comment (liuzhe-lz) [inline src/webui/src/components/TrialsDetail.tsx]: If we default to maximize, this should be `optimize_mode === 'minimize'`

Comment (lvybriage) [reply src/webui/src/components/TrialsDetail.tsx]: So, if user set this filed, it's value must be minimize or maximize.
Ok. I'll change my code.

Comment (liuzhe-lz) [reply src/webui/src/components/TrialsDetail.tsx]: I mean, it can be something else, but in that case we should choose the default behavior.

Comment (lvybriage) [reply src/webui/src/components/TrialsDetail.tsx]: Got it, thanks very much.

File: src/webui/src/components/trial-detail/DefaultMetricPoint.tsx [hunk 2]
```diff
@@ -55,95 +60,195 @@ class DefaultPoint extends React.Component<DefaultPointProps, DefaultPointState>
...
             Object.keys(accSource).map(item => {
                 const items = accSource[item];
-                let temp: Array<number | string>;
-                temp = [items.index, items.acc, JSON.parse(items.searchSpace)];
+                let temp: Array<number | object>;
+                temp = [items.index, items.acc, items.searchSpace];
                 resultList.push(temp);
             });
+            // isViewBestCurve: false show default metric graph
+            // isViewBestCurve: true  show best curve
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/DefaultMetricPoint.tsx]: The comment is outdated

File: src/webui/src/components/trial-detail/DefaultMetricPoint.tsx [hunk 3]
```diff
@@ -153,18 +258,30 @@ class DefaultPoint extends React.Component<DefaultPointProps, DefaultPointState>
...
-        const { defaultSource, accNodata } = this.state;
+        const { defaultSource, accNodata, isViewBestCurve } = this.state;
         return (
             <div>
+                <div className="default-metric">
+                    <div className="position">
+                        {
+                            isViewBestCurve
+                                ?
+                                <span className="bold">Click here to show <span>default curve</span></span>
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/DefaultMetricPoint.tsx]: "to hide optimization curve"

File: src/webui/src/components/trial-detail/DefaultMetricPoint.tsx [hunk 4]
```diff
@@ -153,18 +258,30 @@ class DefaultPoint extends React.Component<DefaultPointProps, DefaultPointState>
...
+                                :
+                                <span className="bold">Click here to show <span>optimization curve</span></span>
+                        }
+                        <Switch defaultChecked={false} onChange={this.loadDefault} />
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/DefaultMetricPoint.tsx]: I think a checkbox with text "optimization curver" will be better.
Not sure.

**Pull request merged**